src/store/actions/socialActions/FacebookActions.js

```diff
@@ -0,0 +1,40 @@
+import axios from 'axios';
+import {
+  FACEBOOK_AUTHENTICATION,
+  FACEBOOK_AUTH_SUCCESS,
+  FACEBOOK_AUTH_FAIL,
+} from 'store/actions/socialTypes';
+import loginSuccess from 'store/actions/loginActions';
+import { isAuthenticated } from 'utils';
+
+const passToken = token => ({
+  user: {
+    auth_token: token
+  }
+});
+
+const facebookSuccess = resp => ({
+  type: FACEBOOK_AUTH_SUCCESS,
+  payload: resp.data.user.auth_token,
+});
+
+const facebookFail = error => ({
+  type: FACEBOOK_AUTH_FAIL,
+  payload: error.response.data.errors,
+});
+
+const LoginFacebook = token => (dispatch) => {
+  dispatch({
+    type: FACEBOOK_AUTHENTICATION,
+  });
+  return axios.post('https://ah-backend-dojo-dev.herokuapp.com/api/auth/facebook/',
+    passToken(token)).then((resp) => {
+    sessionStorage.setItem('ahToken', resp.data.user.auth_token);
+    dispatch(loginSuccess(isAuthenticated()));
+    dispatch(facebookSuccess(resp));
+  }).catch((error) => {
```

Comment:
Please destructure this error object, to something like:
}).catch({ error : response: { data: errors } } }) => {
dispatch(facebookFail(errors));

Reply:
@bisonlou, in my view, I don't think it will apply here because of the structure of my errors but we can sync so that I can understand exactly what you mean